content/ember/v5/deprecate-array-prototype-extensions.md

```diff
@@ -0,0 +1,1265 @@
+  }
+
+  greet(prefix = 'Hello') {
+    return `${prefix} ${this.name}`;
+  }
+}
+
+[new Person('Tom'), new Person('Joe')].map(person => person['greet']?.('Hi')); // ['Hi Tom', 'Hi Joe']
+```
+#### `isAny`
+
+Before
+```js
+const someArray = [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }];
+someArray.isAny('isFruit'); // true
+```
+
+After:
+```js
+const someArray = [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }];
+someArray.any(el => el.isFruit); // true
+```
+
+#### `isEvery`
+Before:
+
+```js
+const someArray = [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }];
+someArray.isEvery('isFruit'); // false
+```
+
+After:
+
+```js
+const someArray = [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }];
+someArray.every(el => el.isFruit); // false
+```
+#### `mapBy`
+
+Before:
+```js
+const someArray = [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }];
+someArray.mapBy('food'); // ['apple', 'beans']
+```
+
+After:
+```js
+const someArray = [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }];
+someArray.map(el => el.food); // ['apple', 'beans']
+```
+
+#### `objectAt`
+
+Before
+```js
+const someArray = [1, 2, 3, undefined];
+someArray.objectAt(1); // 2
+```
+
+After:
+```js
+const someArray = [1, 2, 3, undefined];
+someArray[1] // 2
+```
+
+#### `objectsAt`
+
+Before:
+```js
+const someArray = [1, 2, 3, undefined];
+someArray.objectsAt([1, 2]); // [2, 3]
+```
+
+After:
+```js
+const someArray = [1, 2, 3, undefined];
+[1, 2].map(index => someArray[index]); //[2, 3]
+```
+
+#### `reject`
+
+Before:
+```js
+const someArray = [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }];
+someArray.reject(el => el.isFruit); // [{ food: 'beans', isFruit: false }]
+```
+
+After:
+```js
+const someArray = [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }];
+someArray.filter(el => !el.isFruit); // [{ food: 'beans', isFruit: false }]
+```
+#### `rejectBy`
+
+Before:
+```js
+const someArray = [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }];
+someArray.rejectBy('isFruit'); // [{ food: 'beans', isFruit: false }]
+```
+
+After:
+```js
+const someArray = [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }];
+someArray.filter(el => !el.isFruit); // [{ food: 'beans', isFruit: false }]
+```
+#### `sortBy`
+
+Before:
+```js
+const someArray = [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }];
+someArray.sortBy('food', 'isFruit'); // [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }]
+```
+
+After:
+```js
+const someArray = [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }];
+[...someArray].sort((a, b) => {
+  return a.food?.localeCompare(b.food)
+    ? a.food?.localeCompare(b.food)
+    : a.isFruit - b.isFruit;
+}); // [{ food: 'apple', isFruit: true }, { food: 'beans', isFruit: false }]
+```
+#### `toArray`
```

Comment:
Is this example even needed? I think you would use `toArray` an an array-like object like `A()`, to turn it into a native array. But here we are only dealing with native arrays, so what's the point of `toArray()` usage then?